pkg/kn/commands/source/list.go

```diff
@@ -58,8 +61,15 @@ func NewListCommand(p *commands.KnParams) *cobra.Command {
 			}
 			sourceList, err := dynamicClient.ListSources(filters...)
 			if err != nil {
-				return err
+				if strings.HasPrefix(knerrors.GetError(err).Error(), "403") {
```

Comment:
We are processing the error at https://github.com/knative/client/pull/948/files#diff-1a1baf9d1fe0b4fa3d8d4e515233df48R42 where we're prefixing api status code (403 in case of forbidden) to message.
Though we can change harcoded `403` here and fetch that from lib.